test/runtests.jl

```diff
@@ -491,4 +491,14 @@ end
     @test collect(YAML.load_all(input)) == expected
 end
 
+# issue #236 - special string keys that require enclosure in quotes
+@testset "issue #236" begin
+    for char in [
+        "{", "}", "[", "]", "&", "*", "#", "?", "|", "-", "<", ">",
+        "=", "!", "%", "@", ":", "`", ",", "'", "\"",
+    ]
+        @test YAML.load(YAML.yaml(Dict(string(char) => 0.0))) == Dict(char => 0.0)
```

Comment:
Here I would opt for the shorter `char in "{}[]..."`. Then `char` actually becomes a `Char` and the `@test` line needs a small modification.